sound/soc/intel/boards/sof_da7219.c

```diff
@@ -539,17 +551,15 @@ static int audio_probe(struct platform_device *pdev)
 static const struct platform_device_id board_ids[] = {
 	{
 		.name = "sof_da7219_mx98373",
-		.driver_data = (kernel_ulong_t)(SOF_MAX98373_SPEAKER_AMP_PRESENT |
-					SOF_DA7219_JSL_BOARD),
+		.driver_data = (kernel_ulong_t)(SOF_DA7219_JSL_BOARD),
 	},
 	{
 		.name = "sof_da7219_mx98360a",
-		.driver_data = (kernel_ulong_t)(SOF_MAX98360A_SPEAKER_AMP_PRESENT |
-					SOF_DA7219_JSL_BOARD),
+		.driver_data = (kernel_ulong_t)(SOF_DA7219_JSL_BOARD),
 	},
 	{
 		.name = "adl_mx98360_da7219",
-		.driver_data = (kernel_ulong_t)(SOF_MAX98360A_SPEAKER_AMP_PRESENT),
+		.driver_data = (kernel_ulong_t)(0),
```

Comment:
so do we need this quirk at all then?

Reply:
what happens if we just don't add .driver_data and add a comment to that effect?

Reply:
unsigned long board_quirk = 0;
...
	if (pdev->id_entry && pdev->id_entry->driver_data)
		board_quirk = (unsigned long)pdev->id_entry->driver_data;

Since driver_data is zero (NULL), board_quirk will be the init value 0. It's working good on my ADL board.